src/handler/AuthHandler.ts

```diff
@@ -25,6 +27,12 @@ export class AuthHandler {
     return decodedJwt;
   }
 
+  async authenticateWithApiKey(key: string): Promise<boolean> {
+    if (!(await apiKeyService.checkIfApiKeyExists(key)))
+      throw new Unauthorized("API key is not valid");
+    return true;
```

Comment:
You can drop the return value here as you dont even need it in [AuthMiddleware](https://github.com/LeoTuet/backend-we-want-to-sleep/blob/61-authentication-for-metrics/src/middleware/AuthMiddleware.ts#L19)